ament_tools/build_types/ament_python.py

```diff
@@ -369,14 +369,18 @@ def _undo_install(self, context):
         if os.path.exists(install_log):
             with open(install_log, 'r') as h:
                 lines = [l.rstrip() for l in h.readlines()]
-            directories = []
+            directories = set()
+            install_space = context.install_space + os.sep
             for line in lines:
                 if os.path.exists(line) and \
-                        line.startswith(context.install_space):
+                        line.startswith(install_space):
```

Comment:
I don't see a problem with that. Important (and missing before) is to include the trialing `/` when doing the `startswith`.